src/components/shared/bullet-list/bullet-list.tsx

```diff
@@ -6,30 +6,71 @@
 import PageBlock from '../page-block/page-block';
 import styles from './bullet-list.module.css';
 import { Form, Button, InputGroup } from 'react-bootstrap';
-import { useContext } from 'react';
-import { EditModeContext } from '../../projects/wbs-details/work-package-container/work-package-container';
+import { useContext, useState, useEffect } from 'react';
+import { FormContext } from '../../projects/wbs-details/work-package-container/work-package-container';
 
 interface BulletListProps {
   title: string;
   headerRight: JSX.Element;
   list: JSX.Element[];
   ordered?: boolean;
   readOnly?: boolean;
+  fieldName?: string;
 }
 
-const BulletList: React.FC<BulletListProps> = ({ title, headerRight, list, ordered, readOnly }) => {
-  const editMode = useContext(EditModeContext);
+const BulletList: React.FC<BulletListProps> = ({
+  title,
+  headerRight,
+  list,
+  ordered,
+  readOnly,
+  fieldName
+}) => {
+  const { editMode, setField } = useContext(FormContext);
+  const [bullets, setBullets] = useState(list);
+  const [newBullet, setNewBullet] = useState('');
+
+  useEffect(() => {
+    setBullets(list);
+  }, [editMode, list]);
+
+  function handleAdd() {
+    setBullets([...bullets, <>{newBullet}</>]);
+  }
+
+  function handleDelete() {}
+
+  function handleChange(fieldName: string, s: string) {
+    setField(fieldName, s);
+  }
+
   const addButton = (
     <InputGroup>
-      <Form.Control type="text" placeholder="Input new bullet here" />
-      <Button variant="success">+</Button>
+      <Form.Control
+        type="text"
+        placeholder="Input new bullet here"
+        onChange={(e) => setNewBullet(e.target.value)}
+      />
+      <Button variant="success" onClick={handleAdd}>
+        +
+      </Button>
     </InputGroup>
   );
-  let listPrepared = list.map((bullet, idx) =>
+
+  let listPrepared = bullets.map((bullet, idx) =>
     editMode && !readOnly ? (
-      <InputGroup>
-        <Form.Control type="text" defaultValue={bullet.props.children} placeholder={bullet.props.children} key={idx} />
-        <Button variant="danger">X</Button>
+      <InputGroup aria-required>
+        <Form.Control
+          required
+          type="text"
+          defaultValue={bullet.props.children}
+          placeholder={bullet.props.children}
+          key={idx}
+          onChange={(e) => handleChange(`${fieldName}${idx}`, e.target.value)}
+        />
+        <Button variant="danger" key={idx} onClick={handleDelete}>
+          X
+        </Button>
       </InputGroup>
     ) : (
       <li key={idx}>{bullet}</li>
```

Comment:
Same comment and recommended edits as with the `EditableTextInputList`, `readOnly ` is a prop on `Form.Control` that you can use and we can probably take out the remove button rendering when it is read only.